### ISSUE TITLE
portraits

### ISSUE THREAD

**Issue by babut85** [Voland163/TFTV#8]
i understand that i'm going to express a very bold revolutionary idea now, but the way it's done now is terrible -_-
the portrait should not be searched for by some kind of alphanumeric encoding(it is buggy and does not work in places. have you even looked at your logs? i recommend taking a look. who came up with this scheme at all?! kick him out!), and BY THE NAME OF the fighter. if the portrait is NOT found by name, then the internal render of the game is called(and you do not have to draw dozens of avatars, which at this historical stage rarely correspond to the settings of the fighter)

**Comment by Voland163** [comment]
1) I'm not going to kick myself out, thank you very much.
2) If you prefer renders rather than custom portraits, just pick that
option in Mod---Settings.
3) The name of the custom portraits uses tags in a certain order to hook
the picture to a character in the game (see the readme in the folder).
4) You can't use names to identify portraits because that bears no
connection to the tags the character uses. I understand that it looks
really simple if what you want is to, e.g. add a character with your name
and picture for your own game, but the system is not focused on that; the
primary purpose is to provide better looking portraits that resemble the
customization options chosen by the player.

**Comment by babut85** [comment]
the problem is not my preferences, but the fact that the game render takes into account all the features of the 3d model of a particular fighter. he doesn't care how many fighters there are and what their parameters are, he will draw them correctly anyway. and those portraits that come with the mod have little to do with correctness. moreover, the system of their numbering and search has bugs- judging by the logs, the mod does not always look for the name of the portrait that it should be looking for. yes, this bug can be fixed, but it won't help with another problem- the fact that you drew portraits not in the style of the game(i wanted to "finish drawing" by simply taking the rendered ones from the game, but it wasn't that easy :( ), and drew them much less than they should be. and you're not going to solve this problem.
are you saying that you can get the tags of a fighter, but you can't get his name? i can't believe the game developers did that %D you know, if you would just let us use custom portraits along with game renderers, then even with all the bugs, it would make life much easier. but when we have to choose either one or the other, the bugs become critical

**Comment by Voland163** [comment]
Ok, let me explain to you what you wanted to write here instead of what you
actually did:

"I would like to add some custom portraits to my game, but I'm having
trouble getting the portraits assigned to the characters I want. If it's
not too much trouble, maybe you could add a feature where if there are
images with filenames named after the character, they get that portrait
assigned?"

Yes, I think that should be possible.

**Comment by babut85** [comment]
yes, it's a much more elegant solution than you have right now. but this will not solve another problem- the fact that the mod requires a large number of custom portraits(thanks that not all of them. i understand that you did this because you were too lazy to draw them all). i'm trying to explain once again, are two problems here: the first is an ill-conceived portrait naming system, which also has bugs; the second is the need for a large number of these custom portraits. the first problem can be fixed either by fixing the search algorithm, or immediately make it elegant- i.e., search by the name of the fighter. the second problem can be solved by the fact that if the portrait file is not located(and it does not matter how it is searched for- by a bunch of tags or just by name), then an in-game rendering will be used for the portrait

**Comment by babut85** [comment]
that is, you will no longer need a switch in the settings to use in-game rendering portraits or custom ones. it is enough to delete all custom portraits and then all portraits will be in-game rendering

**Comment by babut85** [comment]
there is another small thing that is easily solved - not all characters that can be used in the name of a fighter can also be used in the file name
